libminifi/include/utils/TestUtils.h

```diff
@@ -0,0 +1,80 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <string>
+
+#include "../../test/TestBase.h"
+
+#ifdef WIN32
+#include <direct.h>
+#define GetCurrentDir _getcwd
+#else
+#include <unistd.h>
+#define GetCurrentDir getcwd
+#endif
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace utils {
+
+std::string getCurrentWorkingDir(void) {
+  char buff[FILENAME_MAX];
+  GetCurrentDir(buff, FILENAME_MAX);
+  std::string current_working_dir(buff);
+  return current_working_dir;
+}
+
+std::string createTempDir(TestController* testController) {
```

Comment:
Why do we need this wrapper? Especially why do we do with asserts.

Reply:
I wanted a wrapper that does not require a directory template, but asserts on the new directory being generated.

Reply:
Can it fail anyhow?
I feel that in this case we verify our own testutils, which seems like a bit overprotective.
A lot of tests are going to fail in case we can't create tempdirs.

Reply:
The main issue, is that `createTempDirectory` swallows whatever is returned from `boost::filesystem::create_directory()`
Relevant code:
```c++
create_dir(tempDirectory);  << RETURN VALUE NOT HANDLED
return tempDirectory;
```
From `create_dir()`:
```c++
if(boost::filesystem::create_directory(dir))
{
  return 0;
}
else
{
  return -1;
}
```
Comparison of the interfaces:
| | FileUtils::create_temp_directory() | TestController::createTempDirectory() | utils::createTempDir() |
| ------------- | ------------- | ------------- |  ------------- |
| Needs format? | Yes | Yes | ✅ NO|
| Checks if dir creation successful? | No | No | ✅ Yes|
| Cleans after itself? | No | ✅ Yes | ✅ Yes |